src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFileUtilities.cs

```diff
@@ -26,5 +27,20 @@ public static LockFile GetLockFile(string lockFilePath, Common.ILogger logger)
 
             return lockFile;
         }
+
+        public static LockFile SafeGetLockFile(string lockFilePath, Common.ILogger logger)
```

Comment:
I'm wondering the same. I can see how this minimizes the risk to other code, but overall I think we only want the safe read. Being more robust on these reads seems like a good idea all around.